File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/ride/RideRatingUseCase.java
package com.lattis.ellipse.domain.interactor.ride;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.UseCase;
import com.lattis.ellipse.domain.repository.RideRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 4/11/17.
 */

public class RideRatingUseCase extends UseCase<Boolean> {

    private RideRepository rideRepository;

    private int trip_id;
    private int rating;


    @Inject
    public RideRatingUseCase(ThreadExecutor threadExecutor,
                          PostExecutionThread postExecutionThread,
                          RideRepository rideRepository) {
        super(threadExecutor, postExecutionThread);
        this.rideRepository = rideRepository;
    }

    public RideRatingUseCase withTripId(int trip_id) {
        this.trip_id = trip_id;
        return this;
    }

    public RideRatingUseCase withRating(int rating) {
        this.rating = rating;
        return this;
    }


    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return this.rideRepository.rateRide(trip_id,rating);
    }
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/ride/RideRatingUseCase.kt
package com.lattis.domain.usecase.ride

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.RideRepository
import com.lattis.domain.usecase.base.UseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class RideRatingUseCase @Inject constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    private val rideRepository: RideRepository
) : UseCase<Boolean>(threadExecutor!!, postExecutionThread) {
    private var trip_id = 0
    private var rating = 0
    fun withTripId(trip_id: Int): RideRatingUseCase {
        this.trip_id = trip_id
        return this
    }

    fun withRating(rating: Int): RideRatingUseCase {
        this.rating = rating
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return rideRepository.rateRide(trip_id, rating)
    }

}